torch/nn/qat/modules/conv.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+from torch.nn.modules.conv import Conv2d as NNConv2d
+from torch.nn.modules.utils import _pair
+from torch.quantization.QConfig import default_qat_qconfig
+import torch.nn.functional as F
+
+class Conv2d(NNConv2d):
+    r"""
+    A Conv2d module attached with FakeQuantize modules for both output
+    activation and weight, used for quantization aware training.
+
+    We adopt the same interface as `torch.nn.Conv2d`, please see https://pytorch.org/docs/stable/nn.html?highlight=conv2d#torch.nn.Conv2d
+    for documentation.
+
+    Similar to `torch.nn.Conv2d`, with FakeQuantize modules initialized to
+    default.
+
+    Attributes:
+        observer: fake quant module for output activation, it's called observer
+            to align with post training flow, TODO: rename?
+        weight_fake_quant: fake quant module for weight
+
+    """
+
+    def __init__(self, in_channels, out_channels, kernel_size, stride=1,
+                 padding=0, dilation=1, groups=1,
+                 bias=True, padding_mode='zeros'):
+        super(Conv2d, self).__init__(in_channels, out_channels, kernel_size,
+                                     stride=stride, padding=padding, dilation=dilation,
+                                     groups=groups, bias=bias, padding_mode=padding_mode)
+        # fake quant module for output activation
+        self.observer = default_qat_qconfig.activation()
+        self.weight_fake_quant = default_qat_qconfig.weight()
+
+    def forward(self, input):
+        if self.padding_mode == 'circular':
```

Comment:
We have code dup with the nn.Conv2d module here. An option that can save some code will be to write the body of this call as a function with weight as an argument. nn.conv2d would call:
 compute_conv(self.weight,...) while qat.conv would call: compute_conv(self.weight_fake_quant(self.weight)). This requires changing nn.conv2d code.